build.js

```diff
@@ -117,345 +117,578 @@ var getModuleConfig = function(api, app) {
+          xml = replaceTextBetween(xml, XML_START_PLUGINS_APPLICATION, XML_END_PLUGINS_APPLICATION, manifestXmlApplicationStr);
+          return fs.writeFileAsync(manifestXml, xml, 'utf-8');
+        } else {
+          logger.log('No plugin XML to inject');
         }
+      })
+      // read and apply plugins to tealeaf build.gradle
+      .then(function () {
 
-        xml = replaceTextBetween(xml, XML_START_PLUGINS_MANIFEST, XML_END_PLUGINS_MANIFEST, manifestXmlManifestStr);
-        xml = replaceTextBetween(xml, XML_START_PLUGINS_ACTIVITY, XML_END_PLUGINS_ACTIVITY, manifestXmlActivityStr);
-        xml = replaceTextBetween(xml, XML_START_PLUGINS_APPLICATION, XML_END_PLUGINS_APPLICATION, manifestXmlApplicationStr);
-        return fs.writeFileAsync(manifestXml, xml, 'utf-8');
-      } else {
-        logger.log('No plugin XML to inject');
-      }
-    });
-}
-
-var installModuleCode = function (api, app, opts) {
-  var moduleConfig = opts.moduleConfig;
-  var outputPath = opts.outputPath;
-
-  function handleFile(baseDir, filePath, replacer) {
-    var ext = path.extname(filePath);
-    if (ext == '.java' || ext === '.aidl') {
-      return fs.readFileAsync(path.join(baseDir, filePath), 'utf-8')
-        .then(function (contents) {
-          var pkgName = contents.match(/(package[\s]+)([a-z.A-Z0-9]+)/g)[0].split(' ')[1];
-          var pkgDir = pkgName.replace(/\./g, "/");
-          var outFile = path.join(outputPath, "src", pkgDir, path.basename(filePath));
-
-          logger.log("Installing Java package", pkgName, "to", outFile);
-
-          // Run injectionSource section of associated module
-          if (replacer && replacer.length > 0) {
-            for (var jj = 0; jj < replacer.length; ++jj) {
-              var findString = replacer[jj].regex;
-              var keyForReplace = replacer[jj].keyForReplace;
-              var replaceString = app.manifest.android[keyForReplace];
-              if (replaceString) {
-                logger.log(" - Running find-replace for", findString, "->", replaceString, "(android:", keyForReplace + ")");
-                var rexp = new RegExp(findString, "g");
-                contents = contents.replace(rexp, replaceString);
-              } else {
-                logger.error(" - Unable to find android key for", keyForReplace);
+        return Promise.all([
+          fs.readFileAsync(gradleTealeafBuildFile, 'utf-8')]
+          .concat(readTealeafGradleDevkitPluginsDependenciesXMLFiles)
+        )
+
+          .then(function (results) {
+            var xml = results.shift();
+            if (results && results.length > 0 && xml && xml.length > 0) {
+
+              var tealeafGradleBuild_PLUGINS_DEPS = '';
+              var tealeafGradleBuildStr_PLACEHOLDERS = '';
+              var tealeafGradleBuildStr_ANDROID_PLUGINS = '';
+              var tealeafGradleBuildStr_PATCH = '';
+              var tealeafGradleBuildStr_CUSTOM_SETTINGS = '';
```

Comment:
use camelCase for consistancy and better code readablity

Reply:
Done